blockchain/chain_sync/src/sync.rs

```diff
@@ -134,12 +141,38 @@ impl<DB> ChainSyncer<DB>
 where
     DB: BlockStore,
 {
-    /// Starts syncing process
-    pub async fn sync(mut self) -> Result<(), Error> {
+    pub async fn start(mut self) -> Result<(), Error> {
         self.net_handler.spawn(Arc::clone(&self.peer_manager));
 
-        info!("Bootstrapping peers to sync");
+        while let Some(event) = self.network.receiver.next().await {
+            if let NetworkEvent::Hello { source, message } = &event {
+                info!(
+                    "Message inbound, heaviest tipset cid: {:?}",
+                    message.heaviest_tip_set
+                );
+                if let Ok(fts) = self
+                    .fetch_tipset(
+                        source.clone(),
+                        &TipSetKeys::new(message.heaviest_tip_set.clone()),
+                    )
+                    .await
+                {
+                    if self.inform_new_head(&source.clone(), &fts).await.is_err() {
+                        warn!("Failed to sync with provided tipset",);
+                    };
+                } else {
+                    warn!(
+                        "Failed to fetch full tipset from peer: {} from storage or network",
+                        source,
+                    );
+                }
+            }
+        }
+        Ok(())
+    }
 
+    /// Performs syncing process
+    async fn sync(&mut self, head: Tipset) -> Result<(), Error> {
```

Comment:
I don't see why we need to pass ownership of head here. I just tried it with a reference instead, and it works perfectly